### DISCUSSION TITLE
DST sensitive intl based api

### DISCUSSION THREAD

**Discussion by janat08** [js-fns/date-fns#3082] [Ideas]
I wrote a feature that handles dst stuff in another library like 3 years ago, I thought I could implement it here. There are a bunch of issues about dst and I’m guessing your algorithm isn’t as accurate. I was hoping I could apply for open collective payment, with a advance because the repo seems inactive.

**Comment by kossnocorp** [comment]
Hey! Thank you for the proposal. For starters, we would like to understand what exactly you're trying to fix, could you please provide a test case when date-fns fails to respect DST?

**Comment by janat08** [comment]
I looked into tests, and it seems you respect DST changes. That was not the case when I studied this library like 3 years ago. I can't be asked to open a PR without an advance, do you want a better test coverage since not every function is tested at dst transitions? When I emailed you I pretty much meant to create date-fns-tz with dst sensitivity when your tz change results in dst transition (My dst started 2 hours ago, and you're 3 hours behind changing time to my tz).

**Reply by kossnocorp** [reply]
Sorry, I still don't get what you're trying to solve. If you want the discussion to advance, please provide a test case.

**Reply by janat08** [reply]
DST occurs at b constant:

const a = dayjs.tz('2012-03-11 12:00:00', 'Asia/Almaty').tz('America/New_York').format()
const b = dayjs.tz('2012-03-11 13:00:00', 'Asia/Almaty').tz('America/New_York').format()
const ba = dayjs.tz('2012-03-11 14:00:00', 'Asia/Almaty').tz('America/New_York').format()

This pr didn't get merged into the library. Are you asking for posterity, because I think date-fns-tz will pass these tests. Here's the pr: https://github.com/iamkun/dayjs/pull/608/files#diff-ec5582aae402594aa76d994234e7ad69bf3820c696708f8b26e03c6e43a1a118

**Reply by kossnocorp** [reply]
I still don't understand your proposal. date-fns works in the local timezone, and like you said, if you add a month and it results in a DST, it will produce `01:00` instead of `00:00`, which is the expected result. We cover edge cases when it causes trouble, and unless I'm unaware of a problem, we don't have obvious DST bugs.

That might require understanding how things work, and if you do something wrong, the approach will bite you in the ass, but that's the trade-off we're willing to make so that we can keep code minimal.

**Reply by janat08** [reply]
I believe this issue explains it https://github.com/date-fns/date-fns/issues/571. You're not really adding or subtracting time, but changing timezone.

**Reply by kossnocorp** [reply]
> You're not really adding or subtracting time, but changing timezone.

Can you please elaborate, I really struggle to get the point across?

**Comment by janat08** [comment]
Intl api has like 2 methods which means you need to start doing tricks to
convert tz accurately because there’s no api for it.

**Comment by janat08** [comment]
I don’t actually remember details myself, despite being interested in work.

**Comment by janat08** [comment]
I remember what pissed me of about the API. You can't tell when tsd occurs
using API, so when you want to change locale you don't know what tz offset
you actually need. Thats different to subtracting in a sense that intl API
handles it or something.

**Comment by janat08** [comment]
Anyway in the issue that I linked people seem to be very annoyed when
discovering that these edge cases are not covered. The most voted reply
there is about having date packages to handle edge cases and not being
agile or simple to read (having less code).

**Comment by kossnocorp** [comment]
Sorry if I wasn't clear enough. I want to see clear and concrete plans of what you want to do. Linking old issues won't do. Sorry about that.

I want to see a test case so I can understand the exact problem you want to solve or step-by-step plans, so I can understand what's your proposal.

You're touching on so many topics I'm not willing to discuss right now, from timezones support that we don't have to the library architecture and the tradeoffs we make. I would like to focus on one thing only - your proposal.

Also, please make an effort and post it in a single comment. The previous conversation is impossible to follow.

**Comment by janat08** [comment]
The idea is to add intl api based timezone support even if a package for it
exists, because it doesn't appear to cover all use cases. The primary
motivation for the proposal was that I felt like I managed to manipulate
intl api to cover some of important use cases in date library previously
and that you can afford upfront payment with like 80k in open collective.
Although I should be familiar with problem domain I'm unsure myself that
using a date object is sufficient like with the current plugin.

**Reply by kossnocorp** [reply]
Thank you for the answer. Given that we both don't know precisely what problem you're trying to solve, I shall stop wasting your time. We can't use your service, but I appreciate the effort and the desire to help.
